src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -265,14 +260,50 @@ public Atlas sliceRelations(final Shard initialShard)
     {
         final Time time = Time.now();
         final Atlas partiallySlicedExpandedAtlas = buildExpandedAtlas(initialShard);
-        final String shardName = getShardOrAtlasName(partiallySlicedExpandedAtlas);
-        logger.info(STARTED_TASK_MESSAGE, RELATION_SLICING_TASK, shardName);
+        final Set<Long> relationsForInitialShard = new HashSet<>();
+        final Optional<Atlas> initialShardOptional = this.atlasFetcher.apply(initialShard);
+        if (initialShardOptional.isPresent())
+        {
+            initialShardOptional.get().relations()
+                    .forEach(relation -> relationsForInitialShard.add(relation.getIdentifier()));
+        }
+        else
+        {
+            throw new CoreException(
+                    "Could not get data for initial shard {} during relation slicing!",
+                    initialShard.getName());
+        }
 
-        final RawAtlasSlicer relationSlicer = new RawAtlasRelationSlicer(
-                partiallySlicedExpandedAtlas, initialShard, this.countries,
-                this.countryBoundaryMap);
-        logger.info(COMPLETED_TASK_MESSAGE, RELATION_SLICING_TASK, shardName, time.elapsedSince());
-        return relationSlicer.slice();
+        final Predicate<AtlasEntity> filter = entity ->
+        {
+            if (entity.getType().equals(ItemType.RELATION))
+            {
+                return relationsForInitialShard.contains(entity.getIdentifier());
+            }
+            return true;
+        };
+        final Optional<Atlas> subAtlasOptional = partiallySlicedExpandedAtlas.subAtlas(filter,
+                AtlasCutType.SILK_CUT);
+        if (subAtlasOptional.isPresent())
+        {
+            final Atlas partiallySlicedExpandedAtlasWithExtraRelationsRemoved = subAtlasOptional
+                    .get();
+            final String shardName = getShardOrAtlasName(
+                    partiallySlicedExpandedAtlasWithExtraRelationsRemoved);
+            logger.info(STARTED_TASK_MESSAGE, RELATION_SLICING_TASK, shardName);
+
+            final RawAtlasSlicer relationSlicer = new RawAtlasRelationSlicer(
+                    partiallySlicedExpandedAtlasWithExtraRelationsRemoved, initialShard,
+                    this.countries, this.countryBoundaryMap);
+            logger.info(COMPLETED_TASK_MESSAGE, RELATION_SLICING_TASK, shardName,
+                    time.elapsedSince());
+            return relationSlicer.slice();
+        }
+        else
+        {
+            throw new CoreException("Error while preparing subatlas for relation slicing shard {}!",
```

Comment:
Here I would have a message that says that the sub-ed partiallySlicedExpandedAtlas returned nothing...